engine/evar_kinds.ml

```diff
@@ -21,12 +21,16 @@ type matching_var_kind = FirstOrderPatVar of Id.t | SecondOrderPatVar of Id.t
 
 type subevar_kind = Domain | Codomain | Body
 
+(* maybe this should be a Projection.t *)
+type is_missing_record_field = { fieldname : Constant.t; recordname : Names.inductive }
+
 type t =
   | ImplicitArg of GlobRef.t * (int * Id.t option)
      * bool (** Force inference *)
   | BinderType of Name.t
   | NamedHole of Id.t (* coming from some ?[id] syntax *)
-  | QuestionMark of obligation_definition_status * Name.t
+  | QuestionMark of obligation_definition_status * Name.t *
+   is_missing_record_field option
```

Comment:
While you're at it you should probably use a record here. This would prevent a lot of noise in the changes below, and you could also define a default record with everything set to `None` or something and use the design pattern of records using `with` clauses.

Reply:
I'd have written it
```
type question_mark = {
     qm_obligation: obligation_definition_status;
     qm_name: Name.t;
     qm_record_field: is_missing_record_field option;
}                                                                                                                             
```
and probably `type record_field` instead of `type is_missing_record_field`.

Reply:
I am not very fond of the the `qm_` pattern, it tends to get kind of messy; a non-perfect alternative @bollu is is put the record definition inside a module.

Reply:
Using a module here is really heave artillery. This is not a very important type, I don't think it is worth it.